Source/System/Drawing/ColorConverter.cs

```diff
@@ -0,0 +1,433 @@
+                                obj = Color.FromArgb(unchecked((int)(0xFF000000 | (uint)(int)intConverter.ConvertFromString(context, culture, text))));
+                            }
+                        }
+
+                        // Nope.  Parse the RGBA from the text.
+                        //
+                        if (obj == null) {
+                            string[] tokens = text.Split(new char[] {sep});
+                            int[] values = new int[tokens.Length];
+                            for (int i = 0; i < values.Length; i++) {
+                                values[i] = unchecked((int)intConverter.ConvertFromString(context, culture, tokens[i]));
+                            }
+
+                            // We should now have a number of parsed integer values.
+                            // We support 1, 3, or 4 arguments:
+                            //
+                            // 1 -- full ARGB encoded
+                            // 3 -- RGB
+                            // 4 -- ARGB
+                            //
+                            switch (values.Length) {
+                                case 1:
+                                    obj = Color.FromArgb(values[0]);
+                                    break;
+
+                                case 3:
+                                    obj = Color.FromArgb(values[0], values[1], values[2]);
+                                    break;
+
+                                case 4:
+                                    obj = Color.FromArgb(values[0], values[1], values[2], values[3]);
+                                    break;
+                            }
+                            tryMappingToKnownColor = true;
+                        }
+
+                        if ((obj != null) && tryMappingToKnownColor) {
+
+                            // Now check to see if this color matches one of our known colors.
+                            // If it does, then substitute it.  We can only do this for "Colors"
+                            // because system colors morph with user settings.
+                            //
+                            int targetARGB = ((Color)obj).ToArgb();
+
+                            foreach (Color c in Colors.Values) {
+                                if (c.ToArgb() == targetARGB) {
+                                    obj = c;
+                                    break;
+                                }
+                            }
+                        }
+                    }
+#if NETSTANDARD20
+                    if (obj == null)
+                    {
+                        throw new ArgumentException(nameof(value));
+                    }
+#else
+                    if (obj == null) {
+                        throw new ArgumentException(SR.Format(SR.InvalidColor, text));
+                    }
+#endif
```

Comment:
I guess you use this to retain the original code and to mark all changes to the copied code, right?

Reply:
Yes, that's correct.